ArjunaJTS/jts/classes/com/arjuna/ats/internal/jts/context/ContextManager.java

```diff
@@ -296,7 +296,7 @@ public final void purgeActions (String threadId)
 
     public final void purgeActions ()
     {
-        purgeActions(ThreadUtil.getThreadId());
+        purgeActions(ThreadUtil.removeThreadId(Thread.currentThread()));
```

Comment:
I think I already mitigated that with: Object arg = threadId == null ? null : otsCurrent.get(threadId);